README.md

```diff
@@ -1,320 +1,194 @@
+```bash
+tox -f unit --ansible -p auto
 ```
-This field is very simple, and should list folder names, anywhere in the tree, to ignore.
-It does not do specialized glob matching or sub-path matching at this time. Anything living under
-any folder whose name appears in this list will be ignored.
 
-To test with ansible versions 2.7.\*, 2.8.\*, and 2.9.\* across every role and scenario:
-```ini
-[ansible]
-ansible = 2.{7,8,9}
+To run all tests across all available environments:
+
+```bash
+tox --ansible -p auto
 ```
 
-Now the output will look like this:
+Note: The `-p auto` flag will run multiple tests in parallel.
+Note: The specific python interpreter will need to be pre-installed on you system, e.g.:
+
 ```bash
-$ tox -l
-ansible27-lint_all
-ansible27-one
-ansible27-roles-my_role-otherscenario
-ansible27-roles-my_role-somescenario
-ansible27-roles-other_role-basic
-ansible27-roles-other_role-default
-ansible27-roles-other_role-somescenario
-ansible27-two
-ansible28-lint_all
-ansible28-one
-ansible28-roles-my_role-otherscenario
-ansible28-roles-my_role-somescenario
-ansible28-roles-other_role-basic
-ansible28-roles-other_role-default
-ansible28-roles-other_role-somescenario
-ansible28-two
-ansible29-lint_all
-ansible29-one
-ansible29-roles-my_role-otherscenario
-ansible29-roles-my_role-somescenario
-ansible29-roles-other_role-basic
-ansible29-roles-other_role-default
-ansible29-roles-other_role-somescenario
-ansible29-two
-python
+sudo dnf install python3.9
 ```
 
-If you want multiple Python versions, you can also specify that:
+## Configuration
+
+`tox-ansible` can be configured using the `tox.ini` file. The following options are available:
 
 ```ini
 [ansible]
-python = 2.7,3.8
+skip =
+    2.10
+    2.11
 ```
 
-```bash
-$ tox -l
-py27-lint_all
-py27-one
-py27-roles-my_role-otherscenario
-py27-roles-my_role-somescenario
-py27-roles-other_role-basic
-py27-roles-other_role-default
-py27-roles-other_role-somescenario
-py27-two
-py38-lint_all
-py38-one
-py38-roles-my_role-otherscenario
-py38-roles-my_role-somescenario
-py38-roles-other_role-basic
-py38-roles-other_role-default
-py38-roles-other_role-somescenario
-py38-two
-python
+This will skip tests in any environment that uses ansible 2.10 or 2.11. The list of strings are used for a simply string in string comparison of environment names.
+
+## Usage in a CI/CD pipeline
+
+The repo contains a github workflow that can be used in a github actions CI/CD pipeline. The workflow will run all tests across all available environments, unless limited by the `skip` option in the `tox.ini` file.
+
+Each environment will be run in a separate job. The workflow will run all jobs in parallel.
+
+The github matrix is dynamically created by `tox-ansible` using the `--gh-matrix` and `--ansible` flags. The list of environments is converted to a list of entries in json format and added the file specified by the "GITHUB_OUTPUT" environment variable. The workflow will read this file and use it to create the matrix.
```

Comment:
I would like to see the format of the `--gh-matrix` output validated more. Personally when I have code that runs on GHA and uses a container I run it in an `ubuntu-latest` and when it uses vagrant for the driver I run it on a `macos-latest` runner because that has Vagrant support.

I would like to be able to do the same in the future, so documentation of the output values here would permit me to craft my usage of it better.

Reply:
I added a test for the GH json, I'll try to remember to add a README entry for it tomorrow :)